### ISSUE TITLE
How does nnUNet select ROI with unbalanced dataset?

### ISSUE THREAD

**Issue by ai-animate** [MIC-DKFZ/nnUNet#517]
Hi fabian, I found that when processing some datasets with unbalanced classes, the original UNet gives very low softmax probabilities but the nnUNet does not have this problem. 
I think it is the ROI selection part of nnUNet that should give the biggest boost, but there is so much code in nnUNet that I don't have a clue where to start reading, and the documentation doesn't mention this part. What is the ROI selection strategy of nnUNet when dealing with this type of dataset?

**Comment by FabianIsensee** [comment]
Hi, 
just have a look at DataLoader3D. It is implemented in there (oversample foreground). Strategy is very simple:
33% of samples are guaranteed to have at least one foreground class. 67% are sampled randomly
Here is how that is implemented
-  select random training case
-  if not oversample_fg: select random patch from that case
- else select random foreground class that is present in that training case. Then select random voxel from that class and make that the center of the patch (respecting image boundaries of course)
- Done
